src/parser.rs

```diff
@@ -7031,6 +7071,21 @@ impl<'a> Parser<'a> {
         Ok(NamedWindowDefinition(ident, window_spec))
     }
 
+    pub fn parse_create_procedure(&mut self, or_alter: bool) -> Result<Statement, ParserError> {
+        let name = self.parse_object_name()?;
+        let params = self.parse_optional_procedure_parameters()?;
+        self.expect_keyword(Keyword::AS)?;
+        self.expect_keyword(Keyword::BEGIN)?;
+        let statements = self.parse_statements()?;
```

Comment:
If you need special behavior of parse_statements, I think you may have to pass in the parameter here